### ISSUE TITLE
Troubleshoot incorrect database pier correction values

### ISSUE THREAD

**Issue by jmfee-usgs** [usgs/geomag-web-absolutes#252]
Wrong default pier being selected for old observations, and pier correction times are incorrect in the database.

I believe this is due to bad data in the database.  When a new pier was entered, existing mark records were re-referenced; but the data model requires a new mark be added with a new id (even if it's otherwise the same mark information as before).  The observation stores the mark id that was used, causing it to select the wrong pier because there are now 2 piers with the same mark id.

- Document the data model and manual steps required to add a new pier correction.
- Use the attached spreadsheet to verify/correct pier corrections for one problem observatory.
- Verify correcting the pier corrections in the database results in the correct pier being selected for old observations.

**Comment by jmfee-usgs** [comment]
[MagProc Pier Correction Histories - 08-18-2016.pdf](https://github.com/usgs/geomag-baseline-calculator/files/431285/MagProc.Pier.Correction.Histories.-.08-18-2016.pdf)

**Comment by jmfee-usgs** [comment]
Abe and I used the attached PDF to investigate the issues identified with the `BOU` observatory. 

- A new `pier` record was added
- The existing `mark` record was modified to point to the new pier correction, instead of adding a new `mark` record.

This led to the observed behavior of historic observations changing to use the new pier correction value.  Below are some of the SQL commands used to identify the problem on an offline copy of the database, which can be used to fix database as part of #255.

We were able to verify these changes corrected the behavior using the following procedure:
- Before making any database changes, open an old observation.  The new pier correction value is displayed.
- Change the database.
- Open an old observation.  The old pier correction value is displayed.
- Create a new observation.  The new pier correction value is displayed.


- Find mark_id referenced by existing observations, along with observation dates
```sql
SELECT
  o.code,
  from_unixtime(obs.begin) as obs_begin,
  obs.mark_id
FROM observation obs
JOIN observatory o on (o.id = obs.observatory_id)
WHERE o.code='BOU';
```

In this case, every existing observation used the mark id `1`.

- Find the pier/observatory associated with the mark with id `1`
```sql
SELECT
  o.code,
  p.id as pier_id,
  p.name,
  p.correction,
  m.id
FROM mark m
JOIN pier p on (p.id = m.pier_id)
JOIN observatory o on (o.id = p.observatory_id)
WHERE m.id = 1;
```

In this case, the mark had been updated to reference the new pier id `214`

- Find all pier epochs associated with the `BOU` observatory
```sql
SELECT
  o.code,
  p.id,
  p.name,
  FROM_UNIXTIME(p.begin) as pier_begin,
  p.correction
FROM pier p
JOIN observatory o on (o.id = p.observatory_id)
WHERE o.code = 'BOU'
ORDER BY p.name, p.begin;
```

Comparing with the attached PDF, the correction the existing mark record was supposed to reference was pier `20` instead of pier `214`.

- Correct the existing `mark` epoch
```sql
UPDATE mark
SET pier_id = 20,
    end = UNIX_TIMESTAMP('2016-01-01T00:00:00Z')
WHERE id = 1;
```

This repoints the mark associated with existing observations to the pier correction that was used when the observations were created.

- Add a new `mark` epoch associated to the new pier correction
```sql
INSERT INTO mark (pier_id, name, begin, azimuth)
VALUES (214, 'AZ', UNIX_TIMESTAMP('2016-01-01T00:00:00Z'), 199.1383);
```

- Update the new pier epoch to default to the new mark
> NOTE: `LAST_INSERT_ID()` only works when this query is run immediate after
> mark is inserted.

```sql
UPDATE pier
SET default_mark_id = LAST_INSERT_ID()
WHERE pier_id = 214
```

This causes the mark to be automatically selected when the pier is selected in the dropdown.
